File: java/src/main/java/com/mlt/decoder/FloatDecoder.java
package com.mlt.decoder;

import com.mlt.metadata.stream.StreamMetadata;
import java.util.ArrayList;
import java.util.List;
import me.lemire.integercompression.IntWrapper;

public class FloatDecoder {
  private FloatDecoder() {}

  public static List<Float> decodeFloatStream(
      byte[] data, IntWrapper offset, StreamMetadata streamMetadata) {
    var values = DecodingUtils.decodeFloatsLE(data, offset, streamMetadata.numValues());
    var valuesList = new ArrayList<Float>(values.length);
    for (var value : values) {
      valuesList.add(value);
    }
    return valuesList;
  }
}

File: rust/src/decoding/float_decoder.rs
use crate::decoding::decoding_utils::DecodingUtils;
use crate::headers::stream::StreamMetadata;

pub struct FloatDecoder {}
impl FloatDecoder {
    pub fn decode_float_stream(data: &[u8], offset: &mut usize, stream_metadata: &StreamMetadata) -> Vec<f32> {
        let values = DecodingUtils::decode_floats_le(data, offset, stream_metadata.num_values as usize);

        Vec::from(&values[0..values.len()])
    }
}
